### YamlMime:ManagedReference
items:
- uid: System.Threading.ThreadStateException
  id: ThreadStateException
  children:
  - System.Threading.ThreadStateException.#ctor
  - System.Threading.ThreadStateException.#ctor(System.String)
  - System.Threading.ThreadStateException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  - System.Threading.ThreadStateException.#ctor(System.String,System.Exception)
  langs:
  - csharp
  name: ThreadStateException
  nameWithType: ThreadStateException
  fullName: System.Threading.ThreadStateException
  type: Class
  summary: "时，将引发的异常<xref href=&quot;System.Threading.Thread&quot;></xref>无效<xref:System.Threading.Thread.ThreadState*>对于方法调用。</xref:System.Threading.Thread.ThreadState*>"
  remarks: "线程创建后，它是中的至少一个<xref:System.Threading.ThreadState>状态中，直到它终止。</xref:System.Threading.ThreadState> `ThreadStateException`由无法执行请求的操作由于线程的当前状态的方法引发。 例如，尝试重新启动已中止的线程调用<xref:System.Threading.Thread.Start%2A>已终止的线程上引发 ThreadStateException。</xref:System.Threading.Thread.Start%2A>       ThreadStateException 使用 HRESULT COR_E_THREADSTATE，其值 0x80131520。       ThreadStateException 的实例的初始属性值的列表，请参阅<xref:System.Threading.ThreadStateException.%23ctor%2A>构造函数。</xref:System.Threading.ThreadStateException.%23ctor%2A>"
  example:
  - "The following example demonstrates an error that causes the system to throw a `ThreadStateException`.  \n  \n [!code-cpp[ThreadStEx#1](~/add/codesnippet/cpp/t-system.threading.threa_3_1.cpp)]\n [!code-vb[ThreadStEx#1](~/add/codesnippet/visualbasic/t-system.threading.threa_3_1.vb)]\n [!code-cs[ThreadStEx#1](~/add/codesnippet/csharp/t-system.threading.threa_3_1.cs)]  \n  \n This code produces the following output:  \n  \n```  \nIn main. Attempting to restart myThread.  \n    Working thread...  \n    Caught: Thread is running or terminated. Cannot restart.  \n```"
  syntax:
    content: >-
      [System.Runtime.InteropServices.ComVisible(true)]

      public class ThreadStateException : SystemException
  inheritance:
  - System.Exception
  - System.SystemException
  implements: []
  inheritedMembers: []
  platform:
  - net462
- uid: System.Threading.ThreadStateException.#ctor
  id: '#ctor'
  parent: System.Threading.ThreadStateException
  langs:
  - csharp
  name: ThreadStateException()
  nameWithType: ThreadStateException.ThreadStateException()
  fullName: System.Threading.ThreadStateException.ThreadStateException()
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "初始化的新实例<xref href=&quot;System.Threading.ThreadStateException&quot;></xref>使用默认属性的类。"
  remarks: "此构造函数初始化`Message`为系统提供的消息的描述错误，如&quot;请求的操作无法执行由于其当前状态的线程上。&quot;的新实例的属性 此消息将当前系统区域性考虑在内。       下表显示<xref:System.Threading.ThreadStateException>.</xref:System.Threading.ThreadStateException>实例的初始属性的值      |属性 |值 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|`null`.|  |<xref:System.Exception.Message%2A>|本地化的错误消息字符串。 |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public ThreadStateException ();
    parameters: []
  overload: System.Threading.ThreadStateException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Threading.ThreadStateException.#ctor(System.String)
  id: '#ctor(System.String)'
  parent: System.Threading.ThreadStateException
  langs:
  - csharp
  name: ThreadStateException(String)
  nameWithType: ThreadStateException.ThreadStateException(String)
  fullName: System.Threading.ThreadStateException.ThreadStateException(String)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "初始化的新实例<xref href=&quot;System.Threading.ThreadStateException&quot;></xref>与指定的错误消息的类。"
  remarks: "此构造函数初始化`Message`属性的新实例使用`message`参数。       下表显示<xref:System.Threading.ThreadStateException>.</xref:System.Threading.ThreadStateException>实例的初始属性的值      |属性 |值 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|`null`.|  |<xref:System.Exception.Message%2A>|错误消息字符串。 |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public ThreadStateException (string message);
    parameters:
    - id: message
      type: System.String
      description: "解释异常原因的错误消息。"
  overload: System.Threading.ThreadStateException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Threading.ThreadStateException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  id: '#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)'
  parent: System.Threading.ThreadStateException
  langs:
  - csharp
  name: ThreadStateException(SerializationInfo,StreamingContext)
  nameWithType: ThreadStateException.ThreadStateException(SerializationInfo,StreamingContext)
  fullName: System.Threading.ThreadStateException.ThreadStateException(SerializationInfo,StreamingContext)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "初始化的新实例<xref href=&quot;System.Threading.ThreadStateException&quot;></xref>用序列化数据的类。"
  remarks: "在反序列化，以便重建通过流传输的异常对象过程中调用此构造函数。 有关详细信息，请参阅[XML 和 SOAP 序列化](~/add/includes/ajax-current-ext-md.md)。"
  syntax:
    content: protected ThreadStateException (System.Runtime.Serialization.SerializationInfo info, System.Runtime.Serialization.StreamingContext context);
    parameters:
    - id: info
      type: System.Runtime.Serialization.SerializationInfo
      description: "<xref href=&quot;System.Runtime.Serialization.SerializationInfo&quot;> </xref>它保存关于所引发异常的序列化的对象数据。"
    - id: context
      type: System.Runtime.Serialization.StreamingContext
      description: "<xref:System.Runtime.Serialization.StreamingContext>它包含有关源或目标的上下文信息。</xref:System.Runtime.Serialization.StreamingContext>"
  overload: System.Threading.ThreadStateException.#ctor*
  exceptions: []
  platform:
  - net462
- uid: System.Threading.ThreadStateException.#ctor(System.String,System.Exception)
  id: '#ctor(System.String,System.Exception)'
  parent: System.Threading.ThreadStateException
  langs:
  - csharp
  name: ThreadStateException(String,Exception)
  nameWithType: ThreadStateException.ThreadStateException(String,Exception)
  fullName: System.Threading.ThreadStateException.ThreadStateException(String,Exception)
  type: Constructor
  assemblies:
  - mscorlib
  namespace: System.Threading
  summary: "初始化的新实例<xref href=&quot;System.Threading.ThreadStateException&quot;></xref>使用指定的错误消息和对导致此异常的内部异常的引用的类。"
  remarks: "前一个异常的直接结果应包含中的前一个异常的引用引发的异常，<xref:System.Exception.InnerException%2A>属性。</xref:System.Exception.InnerException%2A> <xref:System.Exception.InnerException%2A>属性返回相同的值传递到构造函数中，或`null`如果<xref:System.Exception.InnerException%2A>属性不会提供给构造函数的内部异常值。</xref:System.Exception.InnerException%2A> </xref:System.Exception.InnerException%2A>       下表显示<xref:System.Threading.ThreadStateException>.</xref:System.Threading.ThreadStateException>实例的初始属性的值      |属性 |值 |  |--------------|-----------|  |<xref:System.Exception.InnerException%2A>|内部异常引用。 |  |<xref:System.Exception.Message%2A>|错误消息字符串。 |</xref:System.Exception.Message%2A> </xref:System.Exception.InnerException%2A>"
  syntax:
    content: public ThreadStateException (string message, Exception innerException);
    parameters:
    - id: message
      type: System.String
      description: "解释异常原因的错误消息。"
    - id: innerException
      type: System.Exception
      description: "导致当前异常的异常。 如果`innerException`参数不是<xref uid=&quot;langword_csharp_null&quot; name=&quot;null&quot; href=&quot;&quot;></xref>中, 引发当前异常<xref uid=&quot;langword_csharp_catch&quot; name=&quot;catch&quot; href=&quot;&quot;></xref>处理内部异常的块。"
  overload: System.Threading.ThreadStateException.#ctor*
  exceptions: []
  platform:
  - net462
references:
- uid: System.SystemException
  isExternal: false
  name: System.SystemException
- uid: System.Threading.ThreadStateException.#ctor
  parent: System.Threading.ThreadStateException
  isExternal: false
  name: ThreadStateException()
  nameWithType: ThreadStateException.ThreadStateException()
  fullName: System.Threading.ThreadStateException.ThreadStateException()
- uid: System.Threading.ThreadStateException.#ctor(System.String)
  parent: System.Threading.ThreadStateException
  isExternal: false
  name: ThreadStateException(String)
  nameWithType: ThreadStateException.ThreadStateException(String)
  fullName: System.Threading.ThreadStateException.ThreadStateException(String)
- uid: System.String
  parent: System
  isExternal: true
  name: String
  nameWithType: String
  fullName: System.String
- uid: System.Threading.ThreadStateException.#ctor(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)
  parent: System.Threading.ThreadStateException
  isExternal: false
  name: ThreadStateException(SerializationInfo,StreamingContext)
  nameWithType: ThreadStateException.ThreadStateException(SerializationInfo,StreamingContext)
  fullName: System.Threading.ThreadStateException.ThreadStateException(SerializationInfo,StreamingContext)
- uid: System.Runtime.Serialization.SerializationInfo
  parent: System.Runtime.Serialization
  isExternal: false
  name: SerializationInfo
  nameWithType: SerializationInfo
  fullName: System.Runtime.Serialization.SerializationInfo
- uid: System.Runtime.Serialization.StreamingContext
  parent: System.Runtime.Serialization
  isExternal: true
  name: StreamingContext
  nameWithType: StreamingContext
  fullName: System.Runtime.Serialization.StreamingContext
- uid: System.Threading.ThreadStateException.#ctor(System.String,System.Exception)
  parent: System.Threading.ThreadStateException
  isExternal: false
  name: ThreadStateException(String,Exception)
  nameWithType: ThreadStateException.ThreadStateException(String,Exception)
  fullName: System.Threading.ThreadStateException.ThreadStateException(String,Exception)
- uid: System.Exception
  parent: System
  isExternal: true
  name: Exception
  nameWithType: Exception
  fullName: System.Exception
- uid: System.Threading.ThreadStateException.#ctor*
  parent: System.Threading.ThreadStateException
  isExternal: false
  name: ThreadStateException
  nameWithType: ThreadStateException.ThreadStateException
